File: backend/app/schemas/task.py
"""
任务相关Schema
"""
from pydantic import BaseModel, Field
from typing import Optional, List, Any
from datetime import datetime


class TaskCreate(BaseModel):
    """创建任务请求"""
    product_id: int = Field(..., description="产品ID")
    task_type: str = Field(..., description="任务类型")


class TaskResponse(BaseModel):
    """任务响应"""
    id: int
    task_id: str
    product_id: int
    task_type: str
    status: str
    progress: int
    result: Optional[Any]
    error_message: Optional[str]
    retry_count: int
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    created_by: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TaskListResponse(BaseModel):
    """任务列表响应"""
    items: List[TaskResponse]
    total: int
    page: int
    page_size: int


class TaskLogResponse(BaseModel):
    """任务日志响应"""
    id: int
    task_id: int
    agent_name: str
    log_level: str
    message: str
    extra_data: Optional[Any]
    created_at: datetime

    class Config:
        from_attributes = True


class TaskRetryRequest(BaseModel):
    """任务重试请求"""
    task_id: int = Field(..., description="任务ID")

File: frontend/src/api/task.ts
import { apiClient } from '@/utils/request'

export interface Task {
  id: number
  task_id: string
  product_id: number
  task_type: string
  status: 'pending' | 'running' | 'success' | 'failed' | 'timeout'
  progress: number
  result?: any
  error_message?: string
  retry_count: number
  started_at?: string
  completed_at?: string
  created_by: number
  created_at: string
  updated_at: string
}

export interface TaskListParams {
  page?: number
  page_size?: number
  product_id?: number
  task_type?: string
  status?: string
}

export interface TaskListResponse {
  code: number
  data: {
    items: Task[]
    total: number
    page: number
    page_size: number
  }
}

export interface CreateTaskParams {
  product_id: number
  task_type: string
}

export interface TaskLog {
  id: number
  task_id: number
  agent_name: string
  log_level: string
  message: string
  extra_data?: any
  created_at: string
}

export const taskApi = {
  // 获取任务列表
  getList: (params: TaskListParams) => {
    return apiClient.get<TaskListResponse>('/api/v1/tasks', { params })
  },

  // 获取任务详情
  getDetail: (id: number) => {
    return apiClient.get<{ code: number; data: Task }>(`/api/v1/tasks/${id}`)
  },

  // 创建任务
  create: (data: CreateTaskParams) => {
    return apiClient.post<{ code: number; data: Task }>('/api/v1/tasks', data)
  },

  // 重试任务
  retry: (id: number) => {
    return apiClient.post<{ code: number; data: Task }>(`/api/v1/tasks/${id}/retry`)
  },

  // 获取任务日志
  getLogs: (id: number) => {
    return apiClient.get<{ code: number; data: TaskLog[] }>(`/api/v1/tasks/${id}/logs`)
  },

  // 删除任务
  delete: (id: number) => {
    return apiClient.delete<{ code: number; message: string }>(`/api/v1/tasks/${id}`)
  },
}
